crates/fluvio-smartmodule-derive/src/generator/look_back.rs

```diff
@@ -18,33 +20,30 @@ pub fn generate_look_back_smartmodule(func: &SmartModuleFn) -> TokenStream {
             #[allow(clippy::missing_safety_doc)]
             pub unsafe fn look_back(ptr: *mut u8, len: usize, version: i16) -> i32 {
                 use fluvio_smartmodule::dataplane::smartmodule::{
-                    SmartModuleInput, SmartModuleLookbackErrorStatus,
-                    SmartModuleLookbackRuntimeError, SmartModuleKind, SmartModuleLookbackOutput,
+                    SmartModuleLookbackErrorStatus,
+                    SmartModuleLookbackRuntimeError, SmartModuleKind, SmartModuleLookbackOutput
                 };
-                use fluvio_smartmodule::dataplane::core::{Encoder, Decoder};
-                use fluvio_smartmodule::dataplane::record::{Record, RecordData};
 
                 // DECODING
                 extern "C" {
                     fn copy_records(putr: i32, len: i32);
                 }
 
                 let input_data = Vec::from_raw_parts(ptr, len, len);
-                let mut smartmodule_input = SmartModuleInput::default();
-                if let Err(_err) = Decoder::decode(&mut smartmodule_input, &mut std::io::Cursor::new(input_data), version) {
-                    return SmartModuleLookbackErrorStatus::DecodingBaseInput as i32;
-                }
+
+                #records_code
 
                 let base_offset = smartmodule_input.base_offset();
                 let records_input = smartmodule_input.into_raw_bytes();
-                let mut records: Vec<Record> = vec![];
+
                 if let Err(_err) = Decoder::decode(&mut records, &mut std::io::Cursor::new(records_input), version) {
                     return SmartModuleLookbackErrorStatus::DecodingRecords as i32;
                 };
```

Comment:
This part is not needed here. It breaks the compilation if you try to use `SmartModuleRecord` in the `look_back` function with the message that `SmartModuleRecord` does not impl `Decoder` which is true and it should not be decoded here at all, they are already decoded in the `#records_code` section above.

Reply:
We need to add to new examples also an example of `SmartModuleRecord` used in `look_back` function to check compilation errors.

Reply:
> This part is not needed here. It breaks the compilation if you try to use `SmartModuleRecord` in the `look_back` function with the message that `SmartModuleRecord` does not impl `Decoder` which is true and it should not be decoded here at all, they are already decoded in the `#records_code` section above.

Gotcha! Thanks for pointing it out!

Reply:
Just added a test of look back filter + Timestamps and removed this decoding step!